src/server/game/Globals/ObjectMgr.h

```diff
@@ -861,6 +861,21 @@ struct SceneTemplate
     uint32 ScriptId = 0;
 };
 
+struct StringIdIndex
+{
+    StringIdIndex() = default;
+    StringIdIndex(uint32 index) : Index(index) { }
+
+    uint32 Index = 0;
+
+    explicit operator bool() const { return Index > 0; }
+
+    bool operator==(uint32 a) const;
+    bool operator!=(uint32 a) const { return !(operator==(a)); }
+
+    operator uint32() const { return Index; }
+};
```

Comment:
???

Reply:
Any better idea how I can make it implicit castable to uint32?

Reply:
Naios proposed me to wrap it, I also don’t understand why this is good but well 🤷🏽‍♂️